### PR TITLE
[Cleanup] Unify header guards

### PR DESCRIPTION
**Pull request by Fuzzbawls**
For a while now, we've been using a mish-mash of various header guard styles, with no real consistency project wide. This PR remedies that by setting a strict header guard notation style.

All first-party headers now follow the below guard style:
```cpp
#ifndef PIVX_<PATH_TO>_<FILENAME>_H
#define PIVX_<PATH_TO>_<FILENAME>_H
...
#endif // PIVX_<PATH_TO>_<FILENAME>_H
```
Git subtrees and integrated 3rd party libraries are excluded.

A linter is included so as to prevent future inclusion of include guards that do not adhere to the above standard style.

With this in place, there is more opportunity to "declutter" our file structure quite a bit, **especially** with regards to the GUI files.

### CODE REVIEW

File: test/lint/lint-include-guards.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+HEADER_ID_SUFFIX="_H"
+
+REGEXP_EXCLUDE_FILES_WITH_PREFIX="src/(chiabls/|crypto/ctaes/|crypto/sph_*|ctpl_stl.h|immer/|leveldb/|crc32c/|cxxtimer.h|secp256k1/|tinyformat.h|univalue/)"
+
+EXIT_CODE=0
+for HEADER_FILE in $(git ls-files -- "*.h" | grep -vE "^${REGEXP_EXCLUDE_FILES_WITH_PREFIX}")
+do
+    HEADER_ID_BASE=$(cut -f2- -d/ <<< "${HEADER_FILE}" | sed "s/\.h$//g" | tr / _ | tr - _ | tr "[:lower:]" "[:upper:]")
+    HEADER_ID="${HEADER_ID_PREFIX}${HEADER_ID_BASE}${HEADER_ID_SUFFIX}"
+    if [[ $(grep --count --extended-regexp "^#(ifndef|define|endif //) ${HEADER_ID}" "${HEADER_FILE}") != 3 ]]; then
```
Comment (panleone) [inline test/lint/lint-include-guards.sh:21]:
```suggestion
    if [[ $(grep --count --extended-regexp "^#(ifndef|define|endif //) ${HEADER_ID}($|\s)" "${HEADER_FILE}") != 3 ]]; then
```

**Review by panleone — CHANGES_REQUESTED** [review]
only one small nit:

with the current script we check only that the header guard begins with  `PIVX_<PATH_TO>_<FILENAME>_H`

But something like
`PIVX_<PATH_TO>_<FILENAME>_H2323asd`

would be accepted too.

I suggested a small change to fix the issue

File: test/lint/lint-include-guards.sh (hunk 1 above)
Comment (Fuzzbawls) [reply test/lint/lint-include-guards.sh]: added

**Review by panleone — APPROVED** [review]
tACK 76b04cbfd8c4f7af7cbc46ce447e7d52a634649a

**Review by Liquid369 — APPROVED** [review]
tACK 76b04cbfd8c4f7af7cbc46ce447e7d52a634649a

**Review by Duddino — APPROVED** [review]
ACK 76b04cbfd8c4f7af7cbc46ce447e7d52a634649a

**Pull request merged**